dask_cuda/explicit_comms/dataframe_merge.py

```diff
@@ -18,6 +67,47 @@ async def broadcast(rank, root_rank, eps, df=None):
         return await recv_df(eps[root_rank])
 
 
+def partition_by_hash(df, columns, n_chunks, ignore_index=False):
+    """Splits dataframe into partitions
+
+    The partitions is determined by the hash value of the rows in `columns`.
+
+    Parameters
+    ----------
+    df: DataFrame
+    columns: label or list
+        Column names on which to split the dataframe
+    npartition: int
+        Number of partitions
+    ignore_index : bool, default False
+        Set True to ignore the index of `df`
+
+    Returns
+    -------
+    out: Dict[int, DataFrame]
+        A dictionary mapping integers in {0..npartition} to dataframes.
+    """
+    if df is None:
+        return [None] * n_chunks
+
+    # Hashing `columns` in `df` and assing it to the "_partitions" column
+    df["_partitions"] = partitioning_index(df[columns], n_chunks)
+    # Split `df` based on the hash values in the "_partitions" column
+    try:
+        # For Dask < 2.17 compatibility
+        ret = shuffle_group(df, "_partitions", 0, n_chunks, n_chunks, ignore_index)
```

Comment:
This is not really a part of this PR, I just moved it from dataframe_shuffle.py since it isn't needed anymore.
But I agree, we should cleanup `partition_by_hash()` in a future PR